### DISCUSSION TITLE
Read Variable from CSV fiel

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#23592] [Q&A General] [answered]
Hello, I am trying to change a _vector<Real>_ that I have in my input file with a column of an external CSV file. My vectors are x,y, and z coordinates as follow:

    init_x_coords = '0.010 0.162 0.106 0.010 0.050 0.010 0.150'
    init_y_coords = '0.04175 0.04725 0.05275 0.05825 0.05825 0.05275 0.04725 0.04175'
    init_z_coords = '0.016 0.016 0.016 0.016 0.019 0.019 0.019 0.019'

And I would like to read them from a CSV file with 3 columns. I tried:
```
[VectorPostprocessors]
  [./Input_Params]
    type = CSVReader
    csv_file = Input_Params.csv
    header = true
  [../]
[]
```
and then:

    init_x_coords = Input_Params/x_coord
    init_y_coords = Input_Params/y_coord
    init_z_coords = Input_Params/z_coord

but I receive an error like "cannot convert 'Input_Params/x_coord' to float"

Any other option to do this? Thanks in advance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

What are the types of these paremeters?
A RealVectorValue'd parameter will accept a vector from the input file, but not a VectorPostprocessor vector. 

Guillaume
